speech/gen_dataVals_from_wave_viewer.m

```diff
@@ -107,6 +107,21 @@
             numUserEvents = 0;
         end
 
+        % reorder events first-to-last
+        if numUserEvents >= 2
+            event_times = trialparams.event_params.user_event_times;
+            event_names = trialparams.event_params.user_event_names;
+
+            [~, sortOrder] = sort(event_times(:));
+            if ~isequal(sortOrder', 1:numUserEvents) % events not ordered first-to-last
+                fprintf('Reordering events for trial %d\n', trialnum);
+                trialparams.event_params.user_event_times = event_times(sortOrder);
+                trialparams.event_params.user_event_names = event_names(sortOrder);
+                save(fullfile(trialPath,filename), 'sigmat', 'trialparams');
+            end
+        end
+
+
         try
             bGoodTrial = trialparams.event_params.is_good_trial;
         catch
```

Comment:
Same as above: Using try/catch makes it a bit more opaque why this would fail. Checking for the field makes it more explicit. (Note: as before, this would replace the whole try/catch block but I can't select it all here)

```suggestion
        if isfield(trialparams,'event_params') && isfield(trialparams.eventparams,'is_good_trial')
            bGoodTrial = trialparams.event_params.is_good_trial;
        else
            bGoodTrial = 1; % if field doesn't exist, assume it's good
        end
```